common/lib/xmodule/xmodule/modulestore/tests/test_mixed_modulestore.py

```diff
@@ -2452,18 +2463,125 @@ def test_course_deleted_signal(self, default):
             with self.store.default_store(default):
                 self.assertIsNotNone(self.store.thread_cache.default_store.signal_handler)
 
-                with mock_signal_receiver(SignalHandler.course_deleted) as receiver:
-                    self.assertEqual(receiver.call_count, 0)
+                signal_handler.send.assert_not_called()
 
-                    # Create a course
-                    course = self.store.create_course('org_x', 'course_y', 'run_z', self.user_id)
-                    course_key = course.id
+                # Create a course
+                course = self.store.create_course('org_x', 'course_y', 'run_z', self.user_id)
+                course_key = course.id
 
-                    # Delete the course
-                    course = self.store.delete_course(course_key, self.user_id)
+                # Delete the course
+                course = self.store.delete_course(course_key, self.user_id)
 
-                    # Verify that the signal was emitted
-                    self.assertEqual(receiver.call_count, 1)
+                # Verify that the signal was emitted
+                signal_handler.send.assert_called_with('course_deleted', course_key=course_key)
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_delete_published_item_creates_no_orphans(self, default_store):
+        """
+        Tests delete published item dont create any oprhans in course
+        """
+        self.initdb(default_store)
+        course_locator = self.course.id
+
+        chapter = self.store.create_child(
+            self.user_id, self.course.location, 'chapter', block_id='section_one'
+        )
+
+        sequential = self.store.create_child(
+            self.user_id, chapter.location, 'sequential', block_id='subsection_one'
+        )
+
+        vertical = self.store.create_child(
+            self.user_id, sequential.location, 'vertical', block_id='moon_unit'
+        )
+
+        problem = self.store.create_child(
+            self.user_id, vertical.location, 'problem', block_id='problem'
+        )
+
+        self.store.publish(chapter.location, self.user_id)
+        # Verify that there are no changes
+        self.assertFalse(self._has_changes(chapter.location))
+        self.assertFalse(self._has_changes(sequential.location))
+        self.assertFalse(self._has_changes(vertical.location))
+        self.assertFalse(self._has_changes(problem.location))
+
+        # No oprhans in course
+        course_orphans = self.store.get_orphans(course_locator)
```

Comment:
I think it defaults to `draft-preferred` — so it's going to pull up the draft orphans.